src/js/controller/Setup.js

```diff
@@ -62,11 +65,18 @@ define([
         ];
 
         this.start = function () {
+            _setupFailureTimeout = setTimeout(_setupTimeoutHandler.bind(this), _errorTimeoutDelay);
+
             _.defer(_nextTask);
         };
 
+        function _setupTimeoutHandler(){
+            this.destroy();
+            _error('Setup Timeout Error: Setup took longer than '+_errorTimeoutDelay+' seconds to complete.');
```

Comment:
Call it `_errorTimeoutSeconds` and it will make more sense. Do math in one place not two.